.github/workflows/ci.yml

```
@@ -5,35 +5,74 @@ on:
+          commit_message: "style: auto-format (clang-format)"
+          branch: ${{ github.head_ref || github.ref_name }}
+
   build:
     name: Build and Test
     runs-on: ${{ matrix.os }}
+    needs: format
     strategy:
       fail-fast: false
       matrix:
-        os: [ windows-latest, ubuntu-latest, macos-latest ]
+        os: [ windows-latest ]
```

YAMLlint (1.37.1)

[error] too many spaces inside brackets

(brackets)